lxd/patches.go

```diff
@@ -2695,6 +2696,143 @@ func patchDevicesNewNamingScheme(name string, d *Daemon) error {
 	return nil
 }
 
+func patchStorageApiPermissions(name string, d *Daemon) error {
+	storagePoolsPath := shared.VarPath("storage-pools")
+	err := os.Chmod(storagePoolsPath, 0711)
+	if err != nil {
+		return err
+	}
+
+	pools, err := d.cluster.StoragePools()
+	if err != nil && err == db.ErrNoSuchObject {
+		// No pool was configured in the previous update. So we're on a
+		// pristine LXD instance.
+		return nil
+	} else if err != nil {
+		// Database is screwed.
+		logger.Errorf("Failed to query database: %s", err)
+		return err
+	}
+
+	for _, poolName := range pools {
+		pool, err := storagePoolInit(d.State(), poolName)
+		if err != nil {
+			return err
+		}
+
+		ourMount, err := pool.StoragePoolMount()
+		if err != nil {
+			return err
+		}
+
+		if ourMount {
+			defer pool.StoragePoolUmount()
+		}
+
+		// chmod storage pool directory
+		storagePoolDir := shared.VarPath("storage-pools", poolName)
+		err = os.Chmod(storagePoolDir, 0711)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		// chmod containers directory
+		containersDir := shared.VarPath("storage-pools", poolName, "containers")
+		err = os.Chmod(containersDir, 0711)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		// chmod custom subdir
+		customDir := shared.VarPath("storage-pools", poolName, "custom")
+		err = os.Chmod(customDir, 0711)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		// chmod images subdir
+		imagesDir := shared.VarPath("storage-pools", poolName, "images")
+		err = os.Chmod(imagesDir, 0700)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		// chmod snapshots subdir
+		snapshotsDir := shared.VarPath("storage-pools", poolName, "snapshots")
+		err = os.Chmod(snapshotsDir, 0700)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		// Retrieve ID of the storage pool (and check if the storage pool
+		// exists).
+		poolID, err := d.cluster.StoragePoolGetID(poolName)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		volumes, err := d.cluster.StoragePoolNodeVolumesGetType(storagePoolVolumeTypeCustom, poolID)
+		if err != nil && err != db.ErrNoSuchObject {
+			return err
+		}
+
+		for _, vol := range volumes {
+			volStruct, err := storagePoolVolumeInit(d.State(), poolName, vol, storagePoolVolumeTypeCustom)
+			if err != nil {
+				return err
+			}
+
+			ourMount, err := volStruct.StoragePoolVolumeMount()
+			if err != nil {
+				return err
+			}
+			if ourMount {
+				defer volStruct.StoragePoolVolumeUmount()
+			}
+
+			cuMntPoint := getStoragePoolVolumeMountPoint(poolName, vol)
+			err = os.Chmod(cuMntPoint, 0711)
+			if err != nil && !os.IsNotExist(err) {
+				return err
+			}
+		}
+	}
+
+	cRegular, err := d.cluster.ContainersList(db.CTypeRegular)
```

Comment:
@freeekanayaka that will get us all the containers not just the local ones, right?
What's the equivalent to get the local containers as that's what we should be using here.

Reply:
Correct, this gets us all containers, not just the local ones.

I don't think there's an equivalent for local nodes only, but it should look something like this:

```go
// ContainersNodeList returns the names of all the containers of the given type on the local node.
func (c *Cluster) ContainersNodeList(cType ContainerType) ([]string, error) {
	q := fmt.Sprintf("SELECT name FROM containers WHERE type=? AND node_id=? ORDER BY name")
	inargs := []interface{}{cType,c.nodeID}
	var container string
	outfmt := []interface{}{container}
	result, err := queryScan(c.db, q, inargs, outfmt)
	if err != nil {
		return nil, err
	}

	var ret []string
	for _, container := range result {
		ret = append(ret, container[0].(string))
	}

	return ret, nil
}
```

Reply:
@freeekanayaka, can you please send a patch for this?